### ISSUE TITLE
Unable to access metadata of files downloaded from Storage

### ISSUE THREAD

**Issue by Kilo-Loco** [aws-amplify/amplify-flutter#597]
**Describe the bug**
When uploading a file through Amplify Storage, you are able to set custom metadata for the file being uploaded as shown [here](https://docs.amplify.aws/lib/storage/upload/q/platform/flutter). However, there is no way to access the metadata when the file is downloaded.

**To Reproduce**
1. Upload file with metadata
2. Download the file
3. Attempt to access metadata

**Expected behavior**
Both `DownloadFileResult` from `Storage.downloadFile` and `StorageItem` from `Storage.list` should include a metadata property to access the values saved with the file.

**Comment by fjnoyp** [comment]
Hi @Kilo-Loco the Amplify-Flutter Storage category is built on top of Amplify-Android and Amplify-iOS which currently do not support the feature you requested (ie. their download and list file APIs do not provide metadata is the response objects).  We would need to coordinate with those teams to add that feature into their API before adding it to Amplify-Flutter.

**Comment by HuiSF** [comment]
Hi @Kilo-Loco We've released Amplify Flutter [v1.0.0](https://github.com/aws-amplify/amplify-flutter/releases/tag/v1.0.0). The Storage S3 plugin now provides flexible ways to retrieve your object metdata.

For download operations you can specify `getProperties` flag as true while calling download APIs. e.g.

```dart
final downloadDataOperation = Amplify.Storage.downloadData(
  key: 'my-object-key',
  options: StorageDownloadDataOptions(
    accessLevel: StorageAccessLevel.private,
    pluginOptions: const S3DownloadDataPluginOptions(
      getProperties: true,
    ),
  ),
);

final result = await downloadDataOperation.result;
print('Downloaded bytes: ${result.bytes}');
print('Metadata: ${result.downloadedItem.metadata}');
```
`getProperties` flag is also available in other operation plugin options, such as `uploadFile`, `uploadData`, `copy` and `move`.

In addition, object metadata now is fetch-able without downloading the object using the [new API `getProperties`](https://docs.amplify.aws/lib/storage/get-properties/q/platform/flutter/). Please refer to the linked documentation for details.
